### DISCUSSION TITLE
Switching Wallbox setting to solar/mixed charging

### DISCUSSION THREAD

**Discussion by Andreas-48** [rdmtc/RedMatic#531] [Q&A] [answered]
I managed to read out all relevant data from my S10 house power plant via Node-Red Modbus. But I failed for weeks to write the bit that switches the wallbox (also from E3/DC) between solar charging and mixed charging. I found no examples online for writing a register.
The service center has written to me that the correct register for the wallbox control is 40088. And the write value of the register should be decimal 4101 in mixed mode and rewritten to 4103 for switching to solar charging. So bit one is the relevant one. I have tried everything in vain. Writing whole register values or bits only.

Coud anyone write a flow that writes this bit or the whole register by modbus? That would be really helpful, I'm stuck and out of ideas.
I also did not understand how I can write a "Single Coil" by using FC5 to toggle this bit when I can only specify the register and not which bit I want to write.
HELP!

**Comment by Andreas-48** [comment, marked as answer]
For whatever reason I found the correct combination of parameters when I kept trying after my post. After so many weeks.
I will show the numbers here that others don't have to experiment as long as me:
- address 40087 (thank you very much E3/DC customer support :-(  )
- type fc6
![Write-node](https://user-images.githubusercontent.com/77237972/151673342-86eb030c-e361-4c22-99e8-373f2b0261ad.png)

Then I use a function node to toggle the value:
- use the last read-out of the address 40087, that I wrote to a global variable
- toggle bit 1 and use for msg.payload
![function-node to build write-msg](https://user-images.githubusercontent.com/77237972/151673645-a5669f2f-a8e4-4cf5-aba6-ca0cb8c541c0.png)

It's suddenly as easy as that. :-)

I use a button to display the current mode and toggle it:
![output2](https://user-images.githubusercontent.com/77237972/151673811-e677014e-dde8-45fb-816b-6439de16a8bd.png)

![output](https://user-images.githubusercontent.com/77237972/151673756-43ce6bc8-bcda-4521-a18e-7eda4dffc6f8.png)

**Comment by hobbyquaker** [comment]
Beantwortet (Lösung steht im Thread). Geschlossen.

---
*Written by Claude Fable on behalf of hobbyquaker.*
